### PR TITLE
Karinna Iniguez - Hotel - Octos

### PR DESCRIPTION
**Pull request by karinnainiguez**
# Hotel
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe a design decision you had to make when working on this project. What options were you considering? What helped you make your final decision? | I originally decided to make block a class, during our in class discussion on design.  However, I had to re-evaluate that decision, because I had rooms as a class, which knew if it was reserved on a given date.  Because of that aspect of my code, I finally decided to keep block as an attribute of room instead of creating a class for block and had to remove those empty files (block.rb and block_spec.rb).  
Describe a concept that you gained more clarity on as you worked on this assignment. |  Comparing two arrays became more clear, and through my search, I found out about sets in ruby and what types of comparisons you can do if you turn arrays into sets.
Describe a _nominal test_ that you wrote for this assignment. | In my reservation class spec file, I created a test to see if a new reservation could be created with a start date, an end date (later than start date) and a room. 
Describe an _edge case test_ that you wrote for this assignment. |  An edge cases that I tried in that same reservation class spec file, were testing to make sure that an Error would be raised if the end time entered was before the start time entered. 
How do you feel you did in writing pseudocode first, then writing the tests and then the code? | I think I did ok writing pseudocode, but I think a lot of that work was better because we had a chance to discuss the design and start the pseudocode together in class in groups.  I'll try to apply those steps even when I work alone on design decisions.

### CODE REVIEW

File: lib/room.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      end_date = validate_date(end_date)
+      validate_stay(start_date, end_date)
+      stay = end_date - start_date
+
+      stay.to_i.times do |index|
+        @reserved << start_date + index
+      end
+    end
+
+    def blocked?(date)
```
Comment (kariabancroft) [inline lib/room.rb:41]: Methods that end in the `?` should return true or false

**Comment by kariabancroft** [comment]
# Hotel
## What We're Looking For

Feature | Feedback
---     | ---
**Design** |
Demonstrated classes having a single responsibility | Yes - Nice job separating the date validation logic. 
Demonstrated loose coupling |Yes
Methods demonstrate a good use of encapsulation, inputs and outputs | Yes
Wave 1 requirements |Yes
Wave 2 requirements |Yes
Wave 3 requirements |Yes - I think by putting the block data directly in the room class it ended up becoming more than what it needed to be. It is a bit convoluted now how that block data relates to the reservation. Its also difficult to see how you might be able to see all of the rooms reserved for a given block or something along those lines.
